plugins/woocommerce/src/Internal/TransientFiles/TransientFilesEngine.php

```diff
@@ -0,0 +1,496 @@
+		 * @return string The actual directory to use for storing transient files.
+		 *
+		 * @since 8.5.0
+		 */
+		$transient_files_directory = apply_filters( 'woocommerce_transient_files_directory', $transient_files_directory );
+
+		$realpathed_transient_files_directory = $this->legacy_proxy->call_function( 'realpath', $transient_files_directory );
+		if ( false === $realpathed_transient_files_directory ) {
+			throw new Exception( "The base transient files directory doesn't exist: $transient_files_directory" );
+		}
+
+		return untrailingslashit( $realpathed_transient_files_directory );
+	}
+
+	/**
+	 * Create a transient file.
+	 *
+	 * @param string     $file_contents The contents of the file.
+	 * @param string|int $expiration_date A string representing the expiration date formatted as "yyyy-mm-dd", or a number representing the expiration date as a timestamp (the time of day part will be ignored).
+	 * @return string The name of the transient file created (without path information).
+	 * @throws \InvalidArgumentException Invalid expiration date (wrongly formatted, or it's a date in the past).
+	 * @throws \Exception The directory to store the file doesn't exist and can't be created.
+	 */
+	public function create_transient_file( string $file_contents, $expiration_date ): string {
+		if ( is_numeric( $expiration_date ) ) {
+			$expiration_date = gmdate( 'Y-m-d', $expiration_date );
+		} elseif ( ! is_string( $expiration_date ) || ! TimeUtil::is_valid_date( $expiration_date, 'Y-m-d' ) ) {
+			$expiration_date = is_scalar( $expiration_date ) ? $expiration_date : gettype( $expiration_date );
+			throw new InvalidArgumentException( "$expiration_date is not a valid date, expected format: year-month-day" );
+		}
+
+		$expiration_date_object = DateTime::createFromFormat( 'Y-m-d', $expiration_date, TimeUtil::get_utc_date_time_zone() );
+		$today_date_object      = new DateTime( $this->legacy_proxy->call_function('gmdate', 'Y-m-d'), TimeUtil::get_utc_date_time_zone() );
+
+		if ( $expiration_date_object < $today_date_object ) {
+			throw new InvalidArgumentException( "The supplied expiration date, $expiration_date, is in the past" );
+		}
+
+		$filename = bin2hex( $this->legacy_proxy->call_function( 'random_bytes', 16 ) );
+
+		$transient_files_directory  = $this->get_transient_files_directory();
+		$transient_files_directory .= '/' . substr( $expiration_date, 0, 10 );
+		if ( ! $this->legacy_proxy->call_function( 'is_dir', $transient_files_directory ) ) {
+			if ( ! $this->legacy_proxy->call_function( 'wp_mkdir_p', $transient_files_directory ) ) {
+				throw new Exception( "Can't create directory: $transient_files_directory" );
+			}
+		}
+		$filepath = $transient_files_directory . '/' . $filename;
+
+		// phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents
+		if( false === $this->legacy_proxy->call_function( 'file_put_contents', $filepath, $file_contents ) ) {
```

Comment:
Why not use the `$wp_filesystem` method here, and avoid adding a linting exception?